mlflow/store/artifact/cloud_artifact_repo.py

```diff
@@ -212,20 +213,30 @@ def _parallelized_download_from_cloud(self, file_size, remote_file_path, local_p
             )
 
             if failed_downloads:
-                new_cloud_creds = self._get_read_credential_infos([remote_file_path])[0]
-                new_signed_uri = new_cloud_creds.signed_uri
-                new_headers = self._extract_headers_from_credentials(new_cloud_creds.headers)
+                num_retries = 3
```

Comment:
@harupy do you think it might be helpful to make this an environment configuration in case we see some adverse behavior? 

The scenario I'm thinking about is: what if we find in another few weeks that there is some failure mechanism for partial download in R2 that tends to resolve 95% of the time after 5 retries, but the success distribution falls to 50% after 3 retries. Or some similar behavior like this. 
Conversely, what if we find that retrying more than once in AWS or Azure is a complete waste of time and some users might want to fail fast if there are connection issues? 

What do you think about a config here that's defaulted to 3?

Reply:
private envs
```
_MLFLOW_MPD_NUM_RETRIES
_MLFLOW_MPD_RETRY_SLEEP_SECONDS
```